### PR TITLE
Improvements in workflow

### PR DESCRIPTION
**Pull request by mvhulten**
The documentation and job submission script are improved (clean up, bug fixes, usability).

### CODE REVIEW

**Comment by mvhulten** [comment]
@s-poll I think this should be reviewed & merged soon.

Then I will try to create some more focussed merge request.

File: README.md [hunk 1]
```diff
@@ -6,56 +6,55 @@ TSMP2 workflow engine for running simulations. The following examples and descri
...
+Instead of setting `BUDGET_ACCOUNT` you may also replace this variable in `ctl/control_tsmp2.sh`
+
 ``` bash
-cd $PROJECT_DIR
-git clone --recurse-submodules https://github.com/HPSCTerrSys/TSMP2_workflow-engine
-wfe_dir=realpath tsmp2_workflow-engine
+cd $PROJECT/$USER
+git clone https://github.com/HPSCTerrSys/TSMP2_workflow-engine
+wfe_dir=$(realpath tsmp2_workflow-engine)
+git submodule init
```
Comment (s-poll) [inline README.md:25]: A `cd wfe_dir` is missing before line 25.

File: README.md [hunk 2]
```diff
@@ -6,56 +6,55 @@ TSMP2 workflow engine for running simulations. The following examples and descri
...
+git submodule update
```
Comment (s-poll) [inline README.md:26]: `git submodule update --init` for just have one line instead of two

File: README.md [hunk 3]
```diff
@@ -6,56 +6,55 @@ TSMP2 workflow engine for running simulations. The following examples and descri
...
 ```
 
 ## Building the model
 
 The TSMP2 ( https://github.com/HPSCTerrSys/TSMP2 ) should be either already compiled (see [ReadMe TSMP2](https://github.com/HPSCTerrSys/TSMP2/blob/master/README.md)) or compiled with the following steps.
 
-0) change directory
 ```bash
 cd ${wfe_dir}/src/TSMP2
+./build_tsmp2.sh --tsmp2_env env/jsc.2024_Intel.sh --icon --eclm --parflow
```
Comment (s-poll) [inline README.md:35]: Please take the default environment in the ReadMe and not mention it explicitly.

File: README.md [hunk 4]
```diff
@@ -6,56 +6,55 @@ TSMP2 workflow engine for running simulations. The following examples and descri
...
+Adjust the components to your purpose.
 
 ## Run experiment
 
-Create and softlink run-directory on SCRATCH
+Create run-directory on SCRATCH
 ``` bash
-export SCRATCH_DIR=/p/scratch/YOUR_PROJECT/$USER/$sim_id
-mkdir -pv $SCRATCH_DIR
-ln -s $SCRATCH_DIR/run run
```
Comment (s-poll) [inline README.md:37]: Why you have removed the generation of the symlink? I would rather remove the run-directory than removing the softlink. In case of no generation of symlink, also the generation of the `$SCRATCH_DIR` would not be necessary.

File: ctl/control_tsmp2.sh [hunk 5]
```diff
@@ -138,9 +138,9 @@ source ${ctl_dir}/utils_tsmp2.sh
 
 # generic sbatch string
 jobgenstring="--export=ALL \
-	      --account=${account} \
+              --account=${account} \
```
Comment (s-poll) [inline ctl/control_tsmp2.sh:141]: What have changed in this line (in web-interface not visible).

File: ctl/control_tsmp2.sh [hunk 6]
```diff
@@ -92,10 +92,10 @@ echo "geo_dir: "${geo_dir}
 # select machine defaults, if not set by user
 if ( [ -z $npnode_u ] | [ -z $partition_u ] ); then
 echo "Take system default for npnode and partition. "
-if [ ${SYSTEMNAME^^} == "JUWELS" ];then
+if [ "${SYSTEMNAME}" == "juwels" ]; then
```
Comment (s-poll) [inline ctl/control_tsmp2.sh:95]: Even though I prefer the lowercase writing, I would stick to the uppercase case writing for consistency reasons.

**Review by s-poll — COMMENTED** [review]
Thanks for your suggestions, here are my comments.

File: README.md (hunk 3 above)
Comment (mvhulten) [reply README.md]: Fixed in 8a31a9b0f46cc663f46b756354db7f0d5170538a.

File: ctl/control_tsmp2.sh (hunk 5 above)
Comment (mvhulten) [reply ctl/control_tsmp2.sh:141]: Yes, it is!  Select the text.

File: README.md (hunk 4 above)
Comment (mvhulten) [reply README.md:37]: I readded it.

I don't understand why a `run` symlink must be created, i.e. why `run` is at two places, but back to that after the merge maybe.

Comment (mvhulten) [reply README.md:37]: I understand now.  User is to remove the `run` directory with a git command as written in the README.

No `rm -f` needed, but of course it would go wrong if the user has some files in there, but she shouldn't.

File: ctl/sim_ctl/sim_config.sh [hunk 7]
```diff
@@ -16,13 +16,12 @@ echo "###"
 # General
 ####################
 
-# create and clean-up sim_dir
+# create a new simulation run directory
 echo "simdir: "$sim_dir
-if [ -d "${sim_dir}" ]; then
+if [ -e "${sim_dir}" ]; then
```
Comment (s-poll) [inline ctl/sim_ctl/sim_config.sh:21]: maybe I miss something, but why did you change `-d` to `-e`? It should be a directory.

File: ctl/sim_ctl/sim_config.sh [hunk 8]
```diff
@@ -16,13 +16,12 @@ echo "###"
...
   mv ${sim_dir} ${sim_dir}_bku$(date '+%Y%m%d%H%M%S')
-  #rm -f ${sim_dir:?}
-fi # -d sim_dir
-mkdir -pv $sim_dir
+fi
+mkdir -p $sim_dir
```
Comment (s-poll) [inline ctl/sim_ctl/sim_config.sh:24]: Just curiosity, why you want to remove the `v`-option

Comment (mvhulten) [reply ctl/sim_ctl/sim_config.sh:24]: Not working in OpenBSD (didn't check POSIX), and the message is not needed for the user, I think.

Neither of these argument are essential, though.

Comment (mvhulten) [reply ctl/sim_ctl/sim_config.sh:24]: I checked newest version of POSIX: [no -v option form mkdir](https://pubs.opengroup.org/onlinepubs/9799919799/), so I propose not to use it.

File: ctl/sim_ctl/sim_config.sh (hunk 7 above)
Comment (mvhulten) [reply ctl/sim_ctl/sim_config.sh:21]: Any type of file in the way will thus be renamed.

Comment (mvhulten) [reply ctl/sim_ctl/sim_config.sh:21]: So concretely, the script would continue without problems, no matter what `$sim_dir` is.

Downside of checking only if it is a directory, the script would fail when it is a regular file.  Worse yet, it would continue with an old (supposedly) directory when it is a symlink.

It is a stronger check and moving anything out of the way (without destroying traces of any potential intentions by the user) is good.

**Comment by mvhulten** [comment]
@kvrigor, @kgoergen, @DCaviedesV, I have been adding to this pull request to get things working properly, but it had to be merged at least a week ago.  I'm adding because changes I'm doing are in the same files, so I want to avoid difficult merges.

Since @s-poll is on holidays now, this would get only worse.  Could one of you have a look at the latest changes starting from the open conversation herein, so it can be merged?

**Comment by mvhulten** [comment]
Approach is now that I add on the same branch.  I spoke it over with Paul and we think that is probably prefered over pulling more people in on this.

**Comment by mvhulten** [comment]
Removed commits fafcc75196e56bcc0a1a0469a01b66216489d215 and above, as discussed / OK with @s-poll.

That commit will be another PR.

**Comment by mvhulten** [comment]
@s-poll, @DCaviedesV, @kvrigor, I forgot to squash the commits!  Should I redo with a squash?

I know how to do that (and can force-push), but is it preferred?  Do we care enough?

The history is *reasonably* clean, but there is at least one commit converting a previous one.

**Comment by DCaviedesV** [comment]
I don't mind it... in fact, I tend to prefer not squashing... fine by me as it is.

**Comment by s-poll** [comment]
Fine by me as it is.
